src/rpm_ostree/mod.rs

```diff
@@ -36,15 +40,29 @@ pub struct Release {
 pub enum Payload {
     /// Represent a pure OSTree checksum
     Checksum(String),
-    /// an OCI image name
-    Pullspec(String),
+    /// an OCI image reference
+    Pullspec(Reference),
+}
+
+/// Optionnal details when staging a new deployement.
+#[derive(Clone, Debug)]
+pub struct StagingOptions {
```

Comment:
This makes sense to me, though good also to go back to individual arg passing once we drop the custom origin bits.